packages/base/src/blockchain.ts

```diff
@@ -278,3 +296,121 @@ export const CellbaseWitness = table(
   },
   ["lock", "message"]
 );
+
+/**
+ * from Transantion defined in  @ckb-lumos/base/lib/api.d.ts
+ * ```
+ * export interface Transaction {
+ *  cellDeps: CellDep[];
+ *  hash?: Hash;
+ *  headerDeps: Hash[];
+ *  inputs: Input[];
+ *  outputs: Output[];
+ *  outputsData: HexString[];
+ *  version: HexNumber;
+ *  witnesses: HexString[];
+ *}
+ * to :
+ * interface TransactionCodecType {
+ *   raw: {
+ *     version: Uint32LE;
+ *     cellDeps: DeCellDepVec;
+ *     headerDeps: Byte32Vec;
+ *     inputs: CellInputVec;
+ *     outputs: CellOutputVec;
+ *     outputsData: BytesVec;
+ *   };
+ *   witnesses: BytesVec;
+ * }
+ * ```
+ * @param data Transantion defined in @ckb-lumos/base/lib/api.d.ts
+ * @returns TransactionCodecType
+ */
+export function transformTransactionCodecType(
+  data: api.Transaction
+): TransactionCodecType {
+  return {
+    raw: {
+      version: data.version,
+      cellDeps: data.cellDeps,
+      headerDeps: data.headerDeps,
+      inputs: data.inputs,
+      outputs: data.outputs,
+      outputsData: data.outputsData,
+    },
+    witnesses: data.witnesses,
+  };
+}
+
+export function deTransformTransactionCodecType(
+  data: TransactionUnpackResultType
+): api.Transaction {
+  return {
+    cellDeps: data.raw.cellDeps.map((cellDep) => {
+      return {
+        outPoint: {
+          txHash: cellDep.outPoint.txHash,
+          index: BI.from(cellDep.outPoint.index).toHexString(),
+        },
+        depType: cellDep.depType,
+      };
+    }),
+    headerDeps: data.raw.headerDeps,
+    inputs: data.raw.inputs.map((input) => {
+      return {
+        previousOutput: {
+          txHash: input.previousOutput.txHash,
+          index: BI.from(input.previousOutput.index).toHexString(),
```

Comment:
```suggestion
          // TODO make an enhancer for number codecs
          index: BI.from(input.previousOutput.index).toHexString(),
```

maybe we can create high-order codecs for number codec here?

```ts
function asHexNumberCodec(numberCodec: 
  | BytesCodec<number, BIish> 
  | BytesCodec<BI, BIish>
): BytesCodec<HexNumber, BIish>
```